lib/ticks-to-tree.js

```diff
@@ -99,71 +112,184 @@ function ticksToTree (ticks, options = {}) {
           type = 'WASM'
         }
       }
-
       if (type && type !== 'JS') name += ' [' + type + (kind ? ':' + kind : '') + ']'
-      return { S, name, value: 0, top: 0 }
+      return { S, name, type, value: 0, top: 0 }
     })
-
     labelInitFrames(stack, regExp)
 
-    addToMergedTree(stack.map(({ S, name, value, top }) => ({ S, name, value, top })))
+    addToMergedTree(trees, stack.map(({ S, name, type, value, top }) => ({ S, name, type, value, top })))
     // mutate original (save another loop over stack + extra objects)
-    addToUnmergedTree(stack)
+    addToUnmergedTree(trees, stack)
   })
 
-  function addToMergedTree (stack) {
-    var lastFrame = null
-    stack.forEach((frame, ix) => {
-      // TODO need an elegant way to place inlined frames into *other* stacks
-      // where the function has disappeared – for now we just remove inlined
-      // stacks from merged graphs
-      if (frame.S > 2) return // skip inlined
-      if (ix > 0) lastFrame.children = lastFrame.children || []
-      const children = (ix === 0) ? merged.children : lastFrame.children
-      const child = children.find(({ name }) => name === frame.name)
-
-      if (child === undefined) children.push(frame)
-      else frame = child
-
-      if (ix === stack.length - 1) frame.top++
-      if (ix === 0) merged.value += 1
-      frame.value++
-
-      lastFrame = frame
+  if (sourceMaps) {
+    await Promise.all(Object.keys(jsLibraries).map(async (jsLibrary) => {
+      return loadSourceMap(jsLibrary, sourceMaps)
+    }))
```

Comment:
This is going to be pretty expensive, this should use p-limit or p-map.

Reply:
Will do!